### PR TITLE
JPEG Encoder: return supported pixel formats for vaQuerySurfaceAttributes

### LINKED ISSUE #60: When vaQuerySurfaceAttributes for jpeg encoder, it doesn't return YUY2 format, which should be handled.
Since commit d540e43 landed, we can query for VAEntrypointEncPicture entrypoint. But it doesn't return YUY2 format, which jpeg encoder is capable of handling.

### PR DESCRIPTION
**Pull request by zzoon**
Since commit d540e43 landed, we can query for VAEntrypointEncPicture
entrypoint. But it doesn't return YUY2 format, which jpeg encoder
is capable of handling.

Fixes #60

### CODE REVIEW

File: src/i965_drv_video.c [hunk 1]
```diff
@@ -5935,6 +5935,12 @@ i965_QuerySurfaceAttributes(VADriverContextP ctx,
                   attribs[i].value.value.i = VA_FOURCC_I010;
                   i++;
                 }
+            } else if (obj_config->entrypoint == VAEntrypointEncPicture) {
```
Comment (scott-d-phillips) [inline src/i965_drv_video.c]: This should just be for JPEG, not all encodes.

File: src/i965_drv_video.c [hunk 2]
```diff
@@ -5935,6 +5935,12 @@ i965_QuerySurfaceAttributes(VADriverContextP ctx,
...
+                attribs[i].type = VASurfaceAttribPixelFormat;
+                attribs[i].value.type = VAGenericValueTypeInteger;
+                attribs[i].flags = VA_SURFACE_ATTRIB_GETTABLE | VA_SURFACE_ATTRIB_SETTABLE;
+                attribs[i].value.value.i = VA_FOURCC_YUY2;
```
Comment (scott-d-phillips) [inline src/i965_drv_video.c]: I think we're also missing Y8, UYVY, and RGB. I think adding all these will also necessitate increasing I965_MAX_SURFACE_ATTRIBUTES.

Comment (seanvk) [inline src/i965_drv_video.c]: Yes, we need to include all the formats supported for JPEG Enc.

**Comment by zzoon** [comment]
Patch updated for support YUY2, UYVY, YV16, Y800 (not RGBA as discussion on slack)

**Pull request merged**